lib/citizens_advice_form_builder/elements/error_summary.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module CitizensAdviceFormBuilder
+  module Elements
+    class ErrorSummary < Base
+      def render
+        component = CitizensAdviceComponents::ErrorSummary.new
+        component.with_errors(error_messages)
+
+        component.render_in(@template)
+      end
+
+      private
+
+      def error_messages
+        object.errors.group_by_attribute.map do |attr, errors|
+          {
+            href: "##{field_id_for(attr)}-input",
```

Comment:
When there are multiple input fields part of a fieldset - e.g radio, checkboxes, dates, the error summary should scroll to the first input field. Not sure if this is a design system problem or if it can be adjusted here

Reply:
The design system uses some JS code to find the nearest label or legend to focus. It's currently broken for data fields from the looks of things, and probably needs to support other inputs too, maybe.

It's tricky for the error summary code to know if the attribute with the error was driven by a text field or a radio button in order to know which HTML `id` to target.

Reply:
I think the JS code is working, but it uses the href to match to the id of the input field, which in our case would end in `-input` and that's different for non textarea/text input fields.

> It's tricky for the error summary code to know if the attribute with the error was driven by a text field or a radio button in order to know which HTML id to target.

Agree. Will there be a way to do that on the form side of things? We do it for template letters but the example there is quite different since we use CMS data: https://github.com/citizensadvice/public-website/blob/a5304b890c1f0be832e4fca7e364562b795111fa/app/models/template_letters/letter.rb#L58

Reply:
Yeah, good point. I'm pretty sure that the GOVUK form builder gets round this by making the first radio or checkbox use the same `id` as the equivalent text field, so it's consistent.

Whilst it's possible to do what template letters is doing, by having special code which changes the `id` being targeted based on the type of form element, but I'd much prefer find a way to not need to do this.